staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -186,10 +187,23 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 	if len(chain[0].Subject.CommonName) == 0 {
 		return nil, false, nil
 	}
-	return &authenticator.Response{
-		User: &user.DefaultInfo{
-			Name:   chain[0].Subject.CommonName,
-			Groups: chain[0].Subject.Organization,
-		},
-	}, true, nil
+
+	// one tenant per user
+	if (len(chain[0].Subject.Organization)>1) {
+		return nil, false, nil
+	}
+
+	// tenant exists in the cert
+	if (len(chain[0].Subject.Organization)>0) {
+		return &authenticator.Response{
+			User: &user.DefaultInfo{
+				Name:   chain[0].Subject.CommonName,
+				Groups: chain[0].Subject.OrganizationalUnit,
+				Tenant: chain[0].Subject.Organization[0],
```

Comment:
What will happen if there is no "/O" set in the subject?  It seems the new code didn't check this situation.